Link to VS Code debug docs from learn page (#6862)
ballerina-platform/ballerina-dev-website: swan-lake/learn-the-platform/test-document-the-code/debug-ballerina-programs.md

active: debug-ballerina-programs
---

The Ballerina compiler comes in handy to detect syntax and semantic issues that occur in your code when developing large-scale applications with complex logic. However, it is difficult for the compiler to detect runtime errors like logical errors because they occur during the program execution after a successful compilation. This is where the dedicated debugging tooling support of Ballerina becomes important.

## Debugging sessions

> The VS Code extension provides three types of sessions to debug your Ballerina code.

Reviewer: ```suggestion
The Ballerina VSCode extension offers three types of sessions: Program debug sessions, Test debug sessions, and Remote debug sessions. Debugging can be done using CodeLens or configurations in a `launch.json file`. It also provides a Debug Console to view output and supports various debugging scenarios.
```

> The Ballerina VS Code extension offers three types of sessions (i.e., program, test, and remote). Debugging can be done using CodeLens or configurations in a `launch.json file`. It also provides a **Debug Console** to view the output and perform various debugging scenarios.

Reviewer: ```suggestion
The Ballerina VS Code extension offers three types of debugging sessions (i.e., program debugging, test debugging, and remote debugging). Debugging sessions can be initiated using CodeLens or configurations in a `launch.json file`. It also provides a **Debug Console** to view the output and perform various debugging scenarios.
```

>**Info:** For more information on the debugging sessions and methods, go to <a href="https://wso2.com/ballerina/vscode/docs/debug-the-code/debugging-overview/" target="_blank">Debug sessions</a>.

## Debugging configurations

> You can define configuration attributes in the `launch.json` file to apply them on the above debug sessions.

Reviewer: ```suggestion
The Ballerina debugger allows you to create a `launch.json` file with default configurations for debugging Ballerina programs. By following a few steps, you can generate the file and then modify the configurations to suit your needs. These configurations have specific attributes that can be edited to customize the debugging process, so that you can set program arguments, command options, environment variables, run in a separate terminal, etc. Additionally, you can configure remote debugging by specifying the host address and port number. These configurations provide flexibility and control when debugging Ballerina code in VS Code.
```

The VS Code extension provides three types of sessions to debug your Ballerina code.

The Ballerina extension provides multiple options to debug Ballerina applications, and the most convenient way will be using the context-aware debug codelens support.
>**Info:** For more information on the debug sessions, go to <a href="https://wso2.com/ballerina/vscode/docs/debug-the-code/debugging-overview/#debugging-sessions" target="_blank">Debug sessions</a>.

However, if you require launching debug sessions with additional configurations (i.e., program arguments, environment variables), you can use `launch.json` configurations
(For more information, see [Start a debug session with configurations](#start-a-debug-session-with-configurations)).
## Debugging methods

### Start a program debug session
The VS Code extension provides three types of methods to execute the above debug sessions.

Follow the steps below to start a quick debug session using CodeLens.
>**Info:** For more information on the debug methods, go to <a href="https://wso2.com/ballerina/vscode/docs/debug-the-code/debugging-overview/#debugging-methods" target="_blank">Debug methods</a>.

> 1. Open the folder which includes the Ballerina program you want to debug, and open the source file in the editor.

Reviewer: IMO we don't need to have a separate section on debugging methods, as that is already covered under the previous section. Should we talk about the `debug configurations` instead?

Author: Updated.

> The Ballerina VScode extension currently supports many debugging features that are available in the VS Code editor.

Reviewer: ```suggestion
The Ballerina VS Code extension provides a range of powerful debugging features. Developers can set breakpoints with conditions and logpoints, pause and continue program execution for precise inspection, evaluate expressions at runtime, and view call stacks and strands. These features enhance the debugging experience, enabling effective troubleshooting and analysis of Ballerina code.
```
